### PR TITLE
SUBMARINE-965. Users should not have the privilege to create experiments in different namespaces

### PR DESCRIPTION
**Pull request by kevin85421**
### What is this PR for?
In our multi-tenant architecture, users only can create experiments in their namespaces, but the Submarine workbench enables users to specify namespaces.
截圖 2021-08-08 上午10 17 13

This patch aims to 

### What type of PR is it?
[Bug Fix]

### Todos
1. Remove the function `public void setNamespace(String namespace)` in ExperimentMeta.java
2. Update experiment RESTful API (remove the field "namespace" in the JSON file)
3. Add a test case to make sure whether the experiment is in the namespace specified by `ENV_NAMESPACE`.

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-965

### How should this be tested?
**Test1**
* Step1: Use submarine-operator to create a submarine service in the namespace "submarine-user-test"
* Step2: Create an experiment, and check whether the experiment is in the namespace "submarine-user-test" rather than "default".

截圖 2021-08-08 下午11 57 35

**Test2**
* Remove the field "namespace" in the Submarine workbench

### Screenshots (if appropriate)
截圖 2021-08-08 下午11 57 35

截圖 2021-08-09 上午1 19 46

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

**Comment by kevin85421** [comment]
@pingsutw Can you help me review this PR? Thanks!

File: submarine-server/server-core/src/test/java/org/apache/submarine/server/rest/ExperimentRestApiTest.java [hunk 1]
```diff
@@ -224,7 +222,6 @@ private void verifyResult(Experiment experiment, String uid) {
     assertEquals(experimentFinishedTime, experiment.getFinishedTime());
     assertEquals(metaName, experiment.getSpec().getMeta().getName());
     assertEquals(metaFramework, experiment.getSpec().getMeta().getFramework());
-    assertEquals(metaNamespace, experiment.getSpec().getMeta().getNamespace());
```
Comment (pingsutw) [inline submarine-server/server-core/src/test/java/org/apache/submarine/server/rest/ExperimentRestApiTest.java:227]: I think we should assert that the experiment namespace is default. And we should add a new test to make sure the namespace is what we expect if we update `ENV_NAMESPACE`

**Review by pingsutw — COMMENTED** [review]
other LGTM

**Comment by kevin85421** [comment]
@pingsutw Thank you for your review. I have added the assertion to `ExperimentRestApiTest.java`. In addition, I updated the TODO in this pull request description.

The integration test is very slow in "apache/submarine"; thus, I use the CI result of "kevin85421/hadoop-submarine" to ensure the correctness of this patch.

https://github.com/kevin85421/hadoop-submarine/actions/runs/1113309003

**Comment by KUAN-HSUN-LI** [comment]
@kevin85421 I am confused about running `mvn install -DskipTests` again. I think it will take more time in the integration test.

**Comment by kevin85421** [comment]
@KUAN-HSUN-LI The command `mvn install -DskipTests` is used to upload the latest package to the local repository. To elaborate, the module `submarine-test` has a dependency `submarine-server-core`, and thus we need to upload the latest package. On the other hand, without the command, maven will repost errors.

For more details, you can check these two pull requests.
https://github.com/apache/submarine/pull/695/files
https://github.com/apache/submarine/pull/683/files

Thanks!

**Comment by KUAN-HSUN-LI** [comment]
@kevin85421 Thanks for your reply!
At first, I had mentioned this problem therefore I saved the dependency in the cache. I wished it will use the latest dependency but it didn't work as I expect. For the reason to reduce the CI time and stable tests, I think it is useful to build the submarine in the e2e and k8s jobs. 
(i.e. build -> submarine-e2e  =>  submarine-e2e  and  build -> submarine-k8s  =>  submarine-k8s )

**Review by pingsutw — APPROVED** [review]
LGTM

**Pull request closed without merging**